docs/examples/userguide/language_basics/open_file.pyx

```diff
@@ -0,0 +1,9 @@
+from libc.stdio cimport FILE, fopen
+
+
+def main():
+    cdef FILE* p
+    p = fopen("spam.txt", "r")
+    if p == NULL:
+        raise OSError("Couldn't open the spam file")

```

Comment:
Hmm, actually, I take that back. Why not show how to properly handle a file error?

Reply:
`PyErr_SetFromErrnoWithFilenameObject(OSError, "spam.txt")` should work, from `cpython.exc`.

Reply:
`PyErr_SetFromErrnoWithFilenameObject` is not in `cpython.exc` but in `Includes/Depreciated/python2.5.pxd`.
Furthermore, I can't cimport it because of the filename has a dot in it:
```
from Depreciated.python2.5 cimport PyErr_SetFromErrnoWithFilenameObject
```

I get 
```
runTest (__main__.CythonCompileTestCase)
compiling (c) open_file ...
=== Expected: ===


=== Got: ===
2:24: Expected 'import' or 'cimport'
```

What should I do?

Reply:
Or show both :)

Reply:
Ah, hadn't seen your answer yet. Right, the declarations are missing from `cpython/exc.pxd` and should be added there. And I think we should just delete the old (and outdated) py25.pxd file.

Reply:
I have a question. In Python there is often one and only one way do do things. In the documentation, we have many ways to checks for NULL. Which one should we show as reference (I think it's better to be consistent)?

```python
if p == NULL:
if not p:
if p is NULL:
```

Reply:
For pointers, I would rule out `if p == NULL`, simply because it's unpythonic compared to the typical `None` usage in Python. That means that also for pointer comparisons, the docs should always say `p1 is p2` instead of `p1 == p2`.

Whether it should be `if not p` or `if p is NULL`, well, that's less obvious. Personally, I would lean towards `if not p` because I think it avoids comparing to constants that some people might not be used to, and it's really straight forward from a Python perspective. But others might find `if p is NULL` more explicit and readable. Also, when translating code from C to Cython (which isn't all that uncommon), having the `NULL` in there seems like a straight forward choice.

Is it really bad to show both? People might really prefer one or the other, based on personal taste and background, and they should understand that both do the same thing when they read existing code that uses either of the two (or both).

Reply:
I mean, I don't have an opinion on the matter, I'll go with whatever you think is the right way to do it. Let's use both but change the == and !=.